internal/controller/utils.go

```diff
@@ -59,6 +60,31 @@ func verifyObject(spec acpv1.AzureAppConfigurationProviderSpec) error {
 		}
 	}
 
+	if spec.FeatureFlag != nil {
+		if spec.Target.ConfigMapData == nil || spec.Target.ConfigMapData.Type == acpv1.Default || spec.Target.ConfigMapData.Type == acpv1.Properties {
+			return loader.NewArgumentError("spec.target.configMapData", fmt.Errorf("target.configMapData.type must be json or yaml when FeatureFlag is set"))
```

Comment:
ArgumentError already indicate the setting section is `spec.target.configMapData`. So the message should be `configMap data type must be json or yaml when FeatureFlag is set.`